src/server/utility.h

```diff
@@ -0,0 +1,42 @@
+// Copyright 2022, DragonflyDB authors.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+
+#pragma once
+
+#include <boost/fiber/mutex.hpp>
+
+#include "server/common.h"
+
+namespace dfly {
+
+// AggregateValue is a thread safe utility to store the first
+// non-default value.
+template <typename T> struct AggregateValue {
```

Comment:
Maybe server/utility.h is not the best place for this helper. My other options were common and error, but they seem to specialized for utils